pkg/diag/validator/resource.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2020 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validator
+
+import (
+	"fmt"
+
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type Resource interface {
+	// Namespace of the resource.
+	Namespace() string
+
+	// Kind of resource, e.g., service, pod, deployment
+	Kind() string
+	// Name of the resource, e.g., cluster name, node name, persistent volume name, etc.
+	Name() string
+
+	// Status if resource is healthy or not
+	Status() Status
+
+	// Reason if resource is not stable.
+	Reason() string
+}
+
+type resource struct {
+	namespace string
+	kind      string
+	name      string
+	reason    string
+	status    Status
+}
+
+func (r *resource) Kind() string      { return r.kind }
```

Comment:
Why do we need those functions rather than exported Fields on resource?

Reply:
I removed the fields from interface definition. I still have Public functions and kept the individual fields as private.